satpy/readers/viirs_compact.py

```diff
@@ -99,40 +99,35 @@ def __init__(self, filename, filename_info, filetype_info):
                                or (max(abs(self.min_lat), abs(self.max_lat)) > 60))
 
         self.scans = self.h5f["All_Data"]["NumberOfScans"][0]
-        self.geostuff = self.h5f["All_Data"]['VIIRS-%s-GEO_All' % self.ch_type]
+        self.geography = self.h5f["All_Data"]['VIIRS-%s-GEO_All' % self.ch_type]
 
         for key in self.h5f["All_Data"].keys():
             if key.startswith("VIIRS") and key.endswith("SDR_All"):
                 channel = key.split('-')[1]
                 break
 
-        # FIXME:  this supposes  there is  only one  tiepoint zone  in the
-        # track direction
-        self.scan_size = self.h5f["All_Data/VIIRS-%s-SDR_All" %
-                                  channel].attrs["TiePointZoneSizeTrack"].item()
-        self.track_offset = self.h5f["All_Data/VIIRS-%s-SDR_All" %
-                                     channel].attrs["PixelOffsetTrack"]
-        self.scan_offset = self.h5f["All_Data/VIIRS-%s-SDR_All" %
-                                    channel].attrs["PixelOffsetScan"]
+        # This supposes  there is  only one  tiepoint zone  in the track direction
+        channel_path = "All_Data/VIIRS-%s-SDR_All" % channel
+        self.scan_size = self.h5f[channel_path].attrs["TiePointZoneSizeTrack"].item()
+        self.track_offset = self.h5f[channel_path].attrs["PixelOffsetTrack"][()]
+        self.scan_offset = self.h5f[channel_path].attrs["PixelOffsetScan"][()]
 
         try:
-            self.group_locations = self.geostuff[
-                "TiePointZoneGroupLocationScanCompact"][()]
+            self.group_locations = self.geography["TiePointZoneGroupLocationScanCompact"][()]
         except KeyError:
             self.group_locations = [0]
 
-        self.tpz_sizes = da.from_array(self.h5f["All_Data/VIIRS-%s-SDR_All" % channel].attrs["TiePointZoneSizeScan"],
-                                       chunks=1)
+        self.tpz_sizes = da.from_array(self.h5f[channel_path].attrs["TiePointZoneSizeScan"], chunks=1)
         if len(self.tpz_sizes.shape) == 2:
             if self.tpz_sizes.shape[1] != 1:
                 raise NotImplementedError("Can't handle 2 dimensional tiepoint zones.")
             self.tpz_sizes = self.tpz_sizes.squeeze(1)
-        self.nb_tpzs = self.geostuff["NumberOfTiePointZonesScan"]
-        self.c_align = da.from_array(self.geostuff["AlignmentCoefficient"],
-                                     chunks=tuple(self.nb_tpzs))
-        self.c_exp = da.from_array(self.geostuff["ExpansionCoefficient"],
-                                   chunks=tuple(self.nb_tpzs))
-        self.nb_tpzs = da.from_array(self.nb_tpzs, chunks=1)
+        self.nb_tiepoint_zones = self.geography["NumberOfTiePointZonesScan"][()]
```

Comment:
What do these `[()]` do?

Reply:
they return a numpy array instead of a h5py dataset.